Analysis of Hector climate samples vs FaIR samples

ciara-donegan opened the pull request:
**Running Hector**
The added R script runs Hector with 200 sets of parameters, randomly drawn from 2806 sets generated with Matilda. The global mean surface temperature (GMST) and ocean heat content (OHC, calculated from heat flux) are extracted, as those are two of the climate variables used to run FACTS.

**Parameters**
The ECS parameter (equilibrium climate sensitivity, the expected temperature rise if CO2 is doubled) is generally lower for Hector than for FaIR, with values spread out over a wider range.
![ecs_histogram](https://github.com/user-attachments/assets/cc1995f3-258a-4b62-b3cc-476e848987d5)

**Hector and FaIR climate comparison**
Over the period from 2020-2300 (years used in most FACTS experiments), the values for both GMST and OHC tend to start out lower for Hector compared to FaIR, converging more in later years. Hector's OHC values are initially spread over a wider range than FaIR's, which are much more concentrated around one value in 2020 but spread more over time.
![hector_v_fair](https://github.com/user-attachments/assets/243e14fc-a817-4ab9-9754-c62de999f3de)
![hector_v_fair_histograms](https://github.com/user-attachments/assets/4a5554f0-421a-4d1c-8fad-ea29f1213132)

ciara-donegan committed:
Running hector with 200 samples

Script for getting 200 hector runs and comparing them with FaIR results

ciara-donegan committed:
Hector/FaIR comparison plots

Plots added to compare the two model outputs

ciara-donegan committed:
Code and plot updates

New plots created to compare FaIR and Hector climate outputs

ciara-donegan committed:
Minor code update

Slight update to some labels

ciara-donegan committed:
Sample parameters file added

Sample csv file containing 200 random samples from the parameters generated by Matilda

kdorheim commented on analysis/testing_params.R:
```diff
@@ -0,0 +1,250 @@
+library(gridExtra)
+
+# Set working directory
+setwd("C:/Users/done231/OneDrive - PNNL/Documents/GitHub/xploring-facts-data")
```
Instead of hard-coding paths we should be using relative pathways, this let's the code be reproducible regardless of machines, I am happy to talk about best coding practices with you

kdorheim commented on analysis/testing_params.R:
```diff
@@ -0,0 +1,250 @@
+#        put plots in a google doc or slides for future meetings so we can annotate them
+
+# Load in parameter csv
+params_all <- read.csv("data/hector_params.csv")
```
Please use `file.path` it is more robust across platforms.

kdorheim commented on analysis/testing_params.R:
```diff
@@ -0,0 +1,250 @@
+core <- newcore(ini_file)
+
+# Loop through Hector runs with all 200 samples
+# PROBLEM: some samples will error out (Assertion failed: Flux and pool values may not be negative in ?)
```
This is would be a great question to raise at one of our check ins

kdorheim commented on analysis/testing_params.R:
```diff
@@ -0,0 +1,250 @@
+                        value = double(),
+                        units = character(),
+                        sample = integer())
+for (sample in 1:200) {
```
Let's talk about using for loops in R sometime soon

kdorheim commented on analysis/testing_params.R:
```diff
@@ -0,0 +1,250 @@
+                            nrow = 2)
+
+# Get FaIR climate to compare
+nc_gmst <- nc_open("C:/Users/done231/OneDrive - PNNL/Desktop/SLR_output/offline_gsat.nc")
```
Oh! please add talking about one drive and git to our to discuss list 👍

kdorheim commented on analysis/testing_params.R:
```diff
@@ -0,0 +1,250 @@
+ecs_hist <- ggplot(ecs_all, aes(x=ecs, color=model, fill=model)) +
+    geom_histogram(alpha=0.2, position="identity") +
+    ggtitle("Equilibrium Climate Sensitivity Parameter: Hector vs FaIR") +
+    theme_bw()
```
Question? How are you saving the plots to disk? Are you using the Rstudio GUI? or ggsave somewhere?

kdorheim reviewed (changes requested):
Hi Ciara, thanks for opening the PR, I think there are some coding best practices we should probably go over some time soon. Please take a look at the comments and come up with a list of additional questions for our next meeting. 
Thanks

ciara-donegan committed:
FACTS experiments output markdown file

R Markdown file created for plotting outputs from all FACTS experiments. Currently contains one plot of sea level change from the coupling.ssp585 experiment.

kdorheim commented on facts_outputs.Rmd:
````diff
@@ -0,0 +1,59 @@
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+knitr::opts_knit$set(root.dir = file.path("data","outputs","fair_results"))
````
Nice use of relative paths in your script, but setting the working directory to `file.path("data","outputs","fair_results")` will make working with the Hector results when we get to it somewhat challenging. If you set it one directory level up to "outputs" you would have easier access to the hector results when we get them.

kdorheim commented on facts_outputs.Rmd:
```diff
@@ -0,0 +1,59 @@
+## FACTS Outputs
+
+This document compares the outputs of various FACTS experiments, with the original climate input from the FaIR simple climate model and with added input from Hector *(to be added)*.
+
```
For every experiment & SCM, you will need to repeat the same lines of code for opening the nc file, extracting the data, and getting the summary results. Do you think you could write a function to help do this to minimize the amount of code you have to write for the following experiments?

kdorheim commented on facts_outputs.Rmd:
```diff
@@ -0,0 +1,59 @@
+f_slr_df <- as.data.frame(f_slr)
+f_slr_df$year <- f_years
+
+f_slr_long <- f_slr_df %>%
```
Nice reshaping data from wide to long another way to get the max, min, and mean values without having to do the reshaping is using the apply function. This has the potential to speed up your code. 

This isn't a necessary change more of an FYI the following chunk of code will return vectors that could be used to make your `f_slr_stats` data.frame  

```
  min <- apply(X = f_slr, MARGIN = 1, FUN = min) 
  max <- apply(X = f_slr, MARGIN = 1, FUN = max) 
  mean <- apply(X = f_slr, MARGIN = 1, FUN = mean) 
```

kdorheim reviewed (changes requested):
Hi Ciara it is looking good, one mandatory request and two suggestions.

ciara-donegan committed:
Relative pathing

Changed paths to files in testing_params.R to be relative rather than hard-coding

ciara-donegan committed:
Experiment output files added to repository

Experiment output netCDF files added to OUTPUTS folder in repository

ciara-donegan committed:
Updates to plot code

Updated relative pathing and plot saving in testing_params.R

ciara-donegan committed:
tlm.global experiment added to markdown file

tlm.global experiment run with both FaIR and Hector, both with 2,000 samples under SSP 5-8.5.

ciara-donegan committed:
fittedismip.ssp585 experiment run with Hector

fittedismip.ssp585 experiment run with Hector added to markdown file, comparing output with experiment run with FaIR.

ciara-donegan committed:
Updated figures in markdown file

Updated figures and experiment descriptions in markdown file

ciara-donegan committed:
Description update in markdown file

Small change to description of markdown file

ciara-donegan committed:
Hector coupling.ssp585 output added

coupling.ssp585 experiment run with Hector climate data, plot added to markdown file

ciara-donegan committed:
Markdown wording update

Slight update to some wording in markdown file
